src/baf/knot_vector.h

```diff
@@ -56,6 +56,15 @@ class KnotVector {
   virtual KnotSpan GetKnotSpan(ParamCoord param_coord) const;
   virtual size_t GetMultiplicity(ParamCoord param_coord) const;
   virtual size_t GetNumberOfKnots() const;
+  size_t GetNumberOfDifferentKnots() const {
+    size_t number = 1;
+    for (size_t i = 1; i < knots_.size(); ++i) {
+      if (knots_[i].get() > knots_[i - 1].get()) {
```

Comment:
The point is that Clang-Tidy complains about this. I have to remove the checks `hicpp-braces-around-statements` and `readability-braces-around-statements` so that it's fine with the linter. Should I do it like this or not remove the curly braces?

Reply:
See related comment in new commit.